### PR TITLE
Add Skeletyl keyboard

### PR DESCRIPTION
**Pull request by bstiq**
## Description

Add Skeletyl keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

Regards,

### CODE REVIEW

File: keyboards/bastardkb/skeletyl/rules.mk [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```
Comment (filterpaper) [inline keyboards/bastardkb/skeletyl/rules.mk:20]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
RGB_MATRIX_ENABLE = yes     # Enable keyboard RGB matrix (do not use together with RGBLIGHT_ENABLE)
RGB_MATRIX_DRIVER = WS2812  # RGB matrix driver support
```

File: keyboards/bastardkb/skeletyl/skeletyl.c [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "skeletyl.h"
```
Comment (filterpaper) [inline keyboards/bastardkb/skeletyl/skeletyl.c:18]:
```suggestion
#include "skeletyl.h"

#ifdef RGB_MATRIX_ENABLE
led_config_t g_led_config = { {
    // left
    {  0,  5,  6, 11, 12 },
    {  1,  4,  7, 10, 13 },
    {  2,  3,  8,  9, 14 },
    { 15, NO_LED, 16, 17, NO_LED }, \
    // right
    { 18, 23, 24, 29, 30 },
    { 19, 22, 25, 28, 31 },
    { 20, 21, 26, 27, 32 },
    { 33, NO_LED, 34, 35, NO_LED } \
}, {
    // left
    {   0,   0 }, {   0,  21 }, {   0,  42 }, // col 1
    {  25,  42 }, {  25,  21 }, {  25,   0 }, // col 2
    {  50,   0 }, {  50,  21 }, {  50,  42 },
    {  74,  42 }, {  74,  21 }, {  74,   0 },
    {  99,   0 }, {  99,  21 }, {  99,  42 },
    {  50,  64 }, {  74,  64 }, {  99,  64 }, // left thumb cluster
    // right
    { 224,   0 }, { 224,  21 }, { 224,  42 }, // col 10
    { 198,  42 }, { 198,  21 }, { 198,   0 }, // col 9
    { 174,   0 }, { 174,  21 }, { 174,  42 },
    { 149,  42 }, { 149,  21 }, { 149,   0 },
    { 124,   0 }, { 124,  21 }, { 124,  42 },
    { 174,  64 }, { 149,  64 }, { 124,  64 }  // right thumb cluster
}, {
    // left
    4, 4, 4, 4, 4, 4, 4,
    4, 4, 4, 4, 4, 4, 4, 4,
    2, 2, 2,
    // right
    4, 4, 4, 4, 4, 4, 4,
    4, 4, 4, 4, 4, 4, 4, 4,
    2, 2, 2
} };
#endif
```

File: keyboards/bastardkb/skeletyl/config.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+#define DEBOUNCE 5
+
+#define SOFT_SERIAL_PIN D0
+
+#define LOCKING_SUPPORT_ENABLE
+#define LOCKING_RESYNC_ENABLE
+#define F_SCL 400000L
+#define USB_POLLING_INTERVAL_MS 1
+#define MASTER_RIGHT
```
Comment (filterpaper) [inline keyboards/bastardkb/skeletyl/config.h:47]:
```suggestion
#define MASTER_RIGHT

// RGB matrix support
#ifdef RGB_MATRIX_ENABLE
#    define SPLIT_TRANSPORT_MIRROR
#    define DRIVER_LED_TOTAL 36 // Number of LEDs
#    define RGB_MATRIX_SPLIT { 18, 18 }
#    define RGB_MATRIX_MAXIMUM_BRIGHTNESS 50
#    define RGB_MATRIX_STARTUP_VAL RGB_MATRIX_MAXIMUM_BRIGHTNESS
#    ifndef RGB_DISABLE_WHEN_USB_SUSPENDED
#        define RGB_DISABLE_WHEN_USB_SUSPENDED true
#    endif
#    define RGB_MATRIX_KEYPRESSES
#endif
```

**Review by filterpaper — CHANGES_REQUESTED** [review]
Additions to support RGB matrix.

File: keyboards/bastardkb/skeletyl/rules.mk [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+SPLIT_KEYBOARD = yes
+TAP_DANCE_ENABLE = no
+LTO_ENABLE = yes
+
+AUDIO_SUPPORTED = no
+RGB_MATRIX_SUPPORTED = no
```
Comment (drashna) [inline keyboards/bastardkb/skeletyl/rules.mk:28]: Would want to change this, too
```suggestion
RGB_MATRIX_SUPPORTED = yes
```

**Comment by bstiq** [comment]
Hello.

Thank you for the hard work. I don't have a skeletyl on hand to test, but 100% trust you guys.
If you confirmed it's working on a skeletyl (or mini with one column disabled) you can go ahead and merge.

Regards,

File: keyboards/bastardkb/skeletyl/config.h [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+#define RGBLED_SPLIT { 18, 18 }
+#define RGBLIGHT_ANIMATIONS
+
+#define DEBOUNCE 5
+
+#define SOFT_SERIAL_PIN D0
+
+#define LOCKING_SUPPORT_ENABLE
+#define LOCKING_RESYNC_ENABLE
+#define F_SCL 400000L
```
Comment (drashna) [inline keyboards/bastardkb/skeletyl/config.h:45]: This shouldn't need to be set, as it is the default for AVR.

File: keyboards/bastardkb/skeletyl/rules.mk [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```
Comment (drashna) [inline keyboards/bastardkb/skeletyl/rules.mk:20]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/bastardkb/skeletyl/rules.mk (hunk 6 above)
Comment (filterpaper) [reply keyboards/bastardkb/skeletyl/rules.mk:20]: Should `RGBLIGHT_ENABLE = no` line be left in there, for users to be aware that is an option if they prefer a "simpler" lighting solution?

File: keyboards/bastardkb/skeletyl/config.h (hunk 5 above)
Comment (bstiq) [reply keyboards/bastardkb/skeletyl/config.h]: done, thanks!

**Review by drashna — APPROVED** [review]

File: keyboards/bastardkb/skeletyl/skeletyl.c [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+#ifdef RGB_MATRIX_ENABLE
+led_config_t g_led_config = { {
+    // left
+    {  0,  5,  6, 11, 12 },
+    {  1,  4,  7, 10, 13 },
+    {  2,  3,  8,  9, 14 },
+    { 15, NO_LED, 16, 17, NO_LED }, \
```
Comment (fauxpark) [inline keyboards/bastardkb/skeletyl/skeletyl.c:26]:
```suggestion
    { 15, NO_LED, 16, 17, NO_LED },
```

File: keyboards/bastardkb/skeletyl/skeletyl.c [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    // right
+    { 18, 23, 24, 29, 30 },
+    { 19, 22, 25, 28, 31 },
+    { 20, 21, 26, 27, 32 },
+    { 33, NO_LED, 34, 35, NO_LED } \
```
Comment (fauxpark) [inline keyboards/bastardkb/skeletyl/skeletyl.c:31]:
```suggestion
    { 33, NO_LED, 34, 35, NO_LED }
```

File: keyboards/bastardkb/skeletyl/readme.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+A very small keyboard made for ergonomic enthusiasts.
+
+* Keyboard Maintainer: [Bastard Keyboards](https://github.com/Bastardkb/)
+* Hardware Supported: elite-C V4
+* Hardware Availability: [Bastardkb.com](https://bastardkb.com/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make bastardkb/skeletyl:manna-harbour_miryoku
```
Comment (fauxpark) [inline keyboards/bastardkb/skeletyl/readme.md:11]:
```suggestion
    make bastardkb/skeletyl:default
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/bastardkb/skeletyl/readme.md (hunk 9 above)
Comment (bstiq) [reply keyboards/bastardkb/skeletyl/readme.md:11]: Hello,

The default keymap for the skeletyl is a dummy blank one.
It's meant to be used with layouts made for 5x3, Miryoku being the most popular one - this is why by default I recommend to compile with Miryoku layout.

Regards,

Comment (drashna) [reply keyboards/bastardkb/skeletyl/readme.md:11]: from the PR Checklist: 

- default (and via) keymaps should be “pristine”
  - bare minimum to be used as a “clean slate” for another user to develop their own user-specific keymap
  - standard layouts preferred in these keymaps, if possible
  - default keymap should not enable VIA – the VIA integration documentation requires a keymap called via

Eg, the default should be a reasonable default.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**